matview/management/commands/refresh_matviews.py

```diff
@@ -1,24 +1,57 @@
+from threading import Lock
+
 from django.core.management import BaseCommand
-from django.utils import timezone
 from django.db.models.signals import post_save
 
 from matview.models import MatView
 
 
+def synchronized(lock):
+    """ Synchronization decorator. """
+
+    def wrap(f):
+        def newFunction(*args, **kw):
+            lock.acquire()
+            try:
+                return f(*args, **kw)
+            finally:
+                lock.release()
+        return newFunction
+    return wrap
+
+
 class Command(BaseCommand):
     help = "Refresh all materialized views"
 
+    def add_arguments(self, parser):
+        parser.add_argument(
+            '--threads', action='store', default=4, type=int,
+            help="Number of worker threads to use"
+        )
+
     def handle(self, *args, **options):
-        t = timezone.now()
+        matviews = MatView.objects.all()
+        total = matviews.count()
+        n = 1
+        lock = Lock()
 
+        @synchronized(lock)
         def report_update(sender, **kwargs):
-            nonlocal t
+            nonlocal n
 
             instance = kwargs.get('instance')
-            delta = (instance.last_updated - t).seconds
-            t = timezone.now()
+            delta = (instance.last_updated - instance._last_started).microseconds / 1000
```

Comment:
We're checking just below here in case instance is falsey, so maybe we should move this line inside that block in case instance is None or something.